src/jdk.jartool/share/classes/sun/tools/jar/Main.java

```diff
@@ -461,7 +461,12 @@ private void validateAndClose(File tmpfile) throws IOException {
         try {
             if (ok) {
                 if (fname != null) {
-                    Files.move(path, Paths.get(fname), StandardCopyOption.REPLACE_EXISTING);
+                    Path target = Paths.get(fname);
+                    Path parent = target.getParent();
+                    if (parent != null) {
+                        Files.createDirectories(parent);
+                    }
```

Comment:
Would be good to move the creation after validating the arguments as this would fail after creating the temp zip file.  Be good to do via another PR